### ISSUE TITLE
finger13.py self.parent no needed line

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11394]
|<img alt="Wu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Wu reported|
|-|-|
|Trac ID|trac#685|
|Type|enhancement|
|Created|2004-08-23 21:47:54Z|

Thanks, please merge.

Attachments:

* [finger13.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/27c/27cfac7148112eaa048ea53bbeb10cacab631742/9d2179823f05521b9de9874e00b21683a50f79de.py) (1702 bytes) - added by Wu on 2004-08-23 21:47:54Z - 
* [finger13-parent-685.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/27c/27cfac7148112eaa048ea53bbeb10cacab631742/627d583f3c254d32207a6b4bc9485279ec63c24b.patch) (981 bytes) - added by thijs on 2008-07-11 18:39:54Z - Patch against r24259

<details><summary>Searchable metadata</summary>

```
trac-id__685 685
type__enhancement enhancement
reporter__Wu Wu
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1093297674000000 1093297674000000
changetime__1216291549000000 1216291549000000
version__ 
owner__ 
cc__Wu cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Wu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Wu commented|
|-|-|

```
#!html
<pre>
In the 13th example of the finger service howto, inside the second part of the
Twisted tutorial Twisted from Scratch, or The Evolution of Finger, there is a
line after the creation of the class FingerService that is not needed at all:

class FingerService(service.Service):
    def __init__(self, *args, **kwargs):
        self.parent.__init__(self, *args)
        self.users = kwargs

There is no need of calling the parent there, because I assume that in this case
it is like calling the __init__ method again rom inside it.

Dunno if necessary, but I do attach a file without such line that works perfectly.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Assigning to me
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

Reassigning doc bugs to edsuom.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner|
|-|-|

Removed line in the [[finger13-parent-685.patch](None/finger13-parent-685.patch) attached patch], up for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @thijstriemstra|
|-|-|

Please remove the `*args` argument, and the 'finger' argument to `FingerService`

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [therve](#note_5):
> Please remove the `*args` argument, and the 'finger' argument to `FingerService`

See updated attached [[finger13-parent-685.patch](None/finger13-parent-685.patch) patch].

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

Looks good, please apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24279]) Apply finger13-parent-685.patch: Removing some redundant code from the finger13.py example.

Author: Wu, thijs
Reviewer: therve, exarkun
Fixes: [#11394](https://github.com/twisted/twisted/issues/11394)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
